source/read/cursor.txt

```diff
@@ -0,0 +1,157 @@
+
+.. _cpp-cursors-iterate:
+
+Retrieve All Cursor Documents
+-----------------------------
+
+To iterate over the contents of a ``mongocxx::cursor`` instance, use a ``for`` loop.
+
+The following example uses the ``find()`` method to retrieve all documents that have a
+``name`` value of ``"Dunkin' Donuts"``. It then prints each document from the cursor returned
+by the ``find()`` method:
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/read/cursor.cpp
+      :start-after: start-cursor-all
+      :end-before: end-cursor-all
+      :language: cpp
+      :dedent:
+
+   .. output::
+
+      { "_id" : { "$oid" : "..." }, ... "name" : "Dunkin' Donuts", "restaurant_id" : "40379573" }
+      { "_id" : { "$oid" : "..." }, ... "name" : "Dunkin' Donuts", "restaurant_id" : "40363098" }
+      { "_id" : { "$oid" : "..." }, ... "name" : "Dunkin' Donuts", "restaurant_id" : "40395071" }
+      ...
+
+Retrieve Documents Individually
+-------------------------------
+
+To retrieve documents from a cursor individually, call the ``begin()`` method on
+a ``mongocxx::cursor`` instance. This method returns an instance of ``mongocxx::cursor::iterator``
+that points to the first document in the cursor.
+
+The following example finds all documents in a collection that have a ``name`` value
+of ``"Dunkin' Donuts"``. It then prints the first document from the cursor by calling the
+``begin()`` method:
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/read/cursor.cpp
+      :start-after: start-cursor-first
+      :end-before: end-cursor-first
+      :language: cpp
+      :dedent:
+
+   .. output::
+
+      { "_id" : { "$oid" : "..." }, ... "name" : "Dunkin' Donuts", "restaurant_id" : "40379573" }
+      
+Tailable Cursors
+----------------
+
+When querying on a :manual:`capped collection </core/capped-collections/>`, you
+can use a **tailable cursor** that remains open after the client exhausts the
+results in a cursor. To create a tailable cursor, instantiate a ``mongocxx::options::find``
+object and set its ``cursor_type`` field to ``mongocxx::cursor::type::k_tailable``. Then,
+pass your ``mongocxx::options::find`` instance as an argument to the ``find()`` method.
+
+For example, you can create a capped collection called ``vegetables`` that stores
+documents representing vegetables, as shown in the following code:
+
+.. literalinclude:: /includes/read/cursor.cpp
+    :language: cpp
+    :dedent:
+    :start-after: start-capped-coll
+    :end-before: end-capped-coll
+
+The following code uses a tailable cursor to retrieve all documents in the ``vegetables``
+collection but continues to check the cursor for more documents until it retrieves
+three documents:
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/read/cursor.cpp
+      :start-after: start-tailable
+      :end-before: end-tailable
+      :language: cpp
+      :dedent:
+
+   .. output::
+
+      { "_id" : { "$oid" : "..." }, "name" : "cauliflower" }
```

Comment:
Q: The copy talks about retrieving three documents, but the output only shows two documents.

Reply:
Line 137 ("if you insert another document ...") kinda explains that, but let me know if it should be clearer. Basically there are two documents in the collection, so the cursor will retrieve two, but it's a tailable cursor so it stays open even though the original cursor is exhausted. Then if you insert a third document, the cursor will also store that document.

Reply:
Ah, ok, I see. Maybe edit the "continues to check the cursor" line to be something like:

"keeps the cursor open to check for more documents until it retrieves three documents"